js/noms/src/list.js

```diff
@@ -118,20 +165,30 @@ export default class List<T: Value> extends Collection<IndexedSequence<any>> {
     return this.sequence.range(start, end);
   }
 
+  /**
+   * The number of elements in the list.
+   */
   get length(): number {
     return this.sequence.numLeaves;
   }
 }
 
+/**
+ * ListLeafSequence is used for the leaf lists of a list prolly-tree.
+ */
 export class ListLeafSequence<T: Value> extends IndexedSequence<T> {
   get chunks(): Array<Ref<any>> {
     return getValueChunks(this.items);
   }
 
   cumulativeNumberOfLeaves(idx: number): number {
+    // TODO: What does this do? Document me!
```

Comment:
Can't really rename it to _cumulativeNumberOfLeaves. We want _ to mean private and we might enforce this in babel/flow.

I ended up with:


```js
/**
 * This method is for internal use of sequences. It returns how many leaf items there are up to an
 * index within its sequence.
 */
```